src/messages.rs

```diff
@@ -187,10 +189,28 @@ impl RoutingMessage {
 
 #[derive(PartialEq, Eq, Clone, Debug, RustcEncodable, RustcDecodable)]
 pub struct SignedRoutingMessage {
+    // FIXME: The field `encoded_routing_message` should be private to
+    // avoid setting it with data that represent something other than
+    // serialised RoutingMessage. (`signature` should probably be
+    // private as well for similar reason).
```

Comment:
Yes, there is a load of `Pub`statements that need removed, secret keys as well We should make these private with accessors to allow borrowing